android/java/org/chromium/chrome/browser/crypto_wallet/activities/BraveWalletDAppsActivity.java

```diff
@@ -85,29 +78,75 @@ protected void triggerLayoutInflation() {
         mFragment = null;
         if (mActivityType == ActivityType.SIGN_MESSAGE) {
             mFragment = new SignMessageFragment();
-        } else if (mActivityType == ActivityType.SWITCH_ETHEREUM_CHAIN) {
-            mFragment = new SwitchEthereumChainFragment();
         } else if (mActivityType == ActivityType.ADD_TOKEN) {
             mFragment = new AddTokenFragment();
-        } else if (mActivityType == ActivityType.CONNECT_ACCOUNT) {
-            mFragment = new ConnectAccountFragment();
-        }
-
-        if (mFragment != null) {
-            FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
-            ft.replace(R.id.frame_layout, mFragment);
-            ft.commit();
         }
+        showCurrentFragment();
         onInitialLayoutInflationComplete();
     }
 
     @Override
     public void finishNativeInitialization() {
         super.finishNativeInitialization();
-        if (mActivityType == ActivityType.ADD_ETHEREUM_CHAIN) {
-            addEthereumChain();
+        processPendingDappsRequest();
+        if (mFragment != null) {
```

Comment:
I think the best way to avoid that is to create fragment here inside `finishNativeInitialization`. What do you think?

Reply:
we would guarantee that every single fragment has native initialized and don't deal at all with that inside them.

Reply:
That should work good too, Should I proceed with moving all the fragments creation inside `finishNativeInitialization` and change the flow for chain fragment in this PR?

Reply:
yes please, let's do it in that PR, not much changes, move the fragments creation to `finishNativeInitialization` and move everything in fragments that is inside `finishNativeInitialization` to `onCreateView`